Lib/dbm/sqlite3.py

```diff
@@ -32,6 +34,37 @@ def _normalize_uri(path):
         uri = uri.replace("//", "/")
     return uri
 
+class _ThreadLocalSqliteConnection:
+    def __init__(self, uri: str):
+        self._uri = uri
+        self._conn = {}
+
+    def conn(self):
+        thread = current_thread()
+        idt = id(thread)
+        if idt in self._conn:
+            return self._conn[idt]
+        def thread_deleted(_, idt=idt):
+            # When the thread is deleted, remove the local dict.
+            # Note that this is suboptimal if the thread object gets
+            # caught in a reference loop. We would like to be called
+            # as soon as the OS-level thread ends instead.
+            if self._conn is not None:
+                conn = self._conn.pop(idt)
+                conn.close()
+        wrthread = ref(thread, thread_deleted)
+        try:
+            conn = sqlite3.connect(self._uri, autocommit=True, uri=True)
+            self._conn[id(thread)] = conn
```

Comment:
```suggestion
            self._conn[thread.native_id] = conn
```

Hmm why not using native_id?